oclint-reporters/reporters/PMDReporter.cpp

```diff
@@ -25,27 +25,62 @@ class PMDReporter : public Reporter
                            const std::string &category,
                            int pmdPriority)
                 : name(name), category(category), priority(pmdPriority) {}
+
+        void write(std::ostream &out) const
+        {
+            out << "priority=\"" << priority << "\" ";
+            out << "rule=\"" << name << "\" ";
+            out << "ruleset=\"" << category << "\" ";
+        }
     };
 
-    std::string xmlEscape(const std::string &data)
+    void writeHeader(std::ostream &out, std::string version)
     {
-        std::string output;
-        output.reserve(data.size());
-        for (const auto &c : data)
+        out << "<?xml version=\"1.0\" encoding=\"UTF-8\"?>" << std::endl;
+        out << "<pmd version=\"oclint-" << version << "\">";
+    }
+
+    void writeFooter(std::ostream &out)
+    {
+        out << "</pmd>";
+    }
+
+    void writeCompilerError(std::ostream &out, const Violation &compilerError)
+    {
+        auto ruleDescription = PMDRuleDescription("compiler error",
+                                                  "compiler",
```

Comment:
Wondering if it makes sense to put this, `compiler warning`, and `clang static analyzer` all under the `clang` ruleset.